### PR TITLE
[MAD][DATA1018][Lab-Intro-ml] Elena JP

### PR DESCRIPTION
**Pull request by elenajpp**
Este lab me ha parecido bastante util y para una primera toma de contacto con ml. Aun que he tenido un problema con  el drop de la columna events que no tengo muy claro si puedo solucionarlo. 
Un saludo
Elena

### CODE REVIEW

File: module-3/lab-intro-to-ml/your-code/main.ipynb [hunk 1]
```diff
@@ -5470,11 +5470,33 @@
   },
```
Comment (Gobinde43) [inline module-3/lab-intro-to-ml/your-code/main.ipynb]: - Something is wrong here: "Calculate the ratio of missing rows to total rows":
```Python
#You have written down 

# Your code her
a = (len(missing_values) / len(weather))
a
#1.0
#Try to do something like the following:
missing_values.size / austin.size
#0.10310841546626232
```
- Try to not hard-code numbers in formulaes:
```Python
#Instead of 
124/len(weather)*100
#Do the following
missing_values['PrecipitationSumInches'].size / austin['PrecipitationSumInches'].size
```
- This one "What is the largest number of events happened in a single day? Enter your answer in the next cell."
should be solved with something like a query. The key word here is "single day". 
- "Next, populate the actual values in the dummy columns of austin_fixed.", can be done as the following. This will throw the wanted results. 
```Python
for event in event_list:
    austin_fixed[event] = austin_fixed['Events'].str.contains(event).astype(int)
```
- This "In the cell below, convert the Date column values from string to numeric values using toordinal()." can be done as simple as the following. Your code is not working properly, the date column has been fixed to "732098" in every row of the dataframe. 
```Python
from datetime import datetime as dt
austin_fixed['Date'] = austin_fixed['Date'].apply(lambda date_col: dt.strptime(date_col, '%Y-%m-%d').toordinal())
```
- In this one "Your y should be a subset of austin_fixed containing one column TempAvgF." try to not write down column labels one by one, instead do something like the following: 
```Python
response_col_names = list(austin_fixed.columns.values)
response_col_names.remove('TempAvgF')
X = austin_fixed[response_col_names]
y = austin_fixed['TempAvgF']
```
- If you want to play with synthetic data for classification problems, scikit-learn has a function to generate synthetic datasets. https://scikit-learn.org/stable/modules/generated/sklearn.datasets.make_classification.html.
It's very useful to play with different type of classifiers (two label)
```Python
from sklearn.datasets import make_classification
```
- Bonus Challenge 2: Try to complete the lab and train linear SVM as well as rbf one, compare the performance metrics. Previous to training a classifier it's a good idea to plot a scatter plot of your data for each label/class. This can give you a visual idea of how well a linear or not linear classifier will perform in the training step.
